core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -458,14 +459,16 @@ else if (alignment < 0) {
 
     private static KdbTree loadKdbTree(String tableName, Session session, Metadata metadata, SplitManager splitManager, PageSourceManager pageSourceManager)
     {
-        QualifiedObjectName name = toQualifiedObjectName(tableName, session.getCatalog().get(), session.getSchema().get());
-        TableHandle tableHandle = metadata.getTableHandle(session, name)
-                .orElseThrow(() -> new TrinoException(INVALID_SPATIAL_PARTITIONING, format("Table not found: %s", name)));
+        QualifiedObjectName originalQualifiedTableName = toQualifiedObjectName(tableName, session.getCatalog().get(), session.getSchema().get());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalQualifiedTableName);
+        TableHandle tableHandle = redirectionAwareTableHandle.getTableHandle()
+                .orElseThrow(() -> new TrinoException(INVALID_SPATIAL_PARTITIONING, format("Table not found: %s", originalQualifiedTableName)));
+        QualifiedObjectName qualifiedTableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(originalQualifiedTableName);
```

Comment:
@kokosing  Do we need to add an appropriate method in `io.trino.security.AccessControl` for checking whether the KDB tree table can be used?

Reply:
If that would be a case we would need to do it in semantic analyzer.

Is redirected table equivalent to a view with SECURITY INVOKER?

Reply:
In the https://github.com/trinodb/trino/issues/11202#issue-1150518940 has been pointed out that the ACL checks for this kind of operation are currently missing. This is why i was wondering whether it would need to be added.

> Is redirected table equivalent to a view with SECURITY INVOKER?

I'm not versed enough with the spatial join functionality to answer this question. Just for reference for other viewers of this PR regarding `INVOKER` / `DEFINER` topic: https://trino.io/docs/current/sql/create-view.html
@mbasmanova could you please share your thoughts on this question?

Reply:
> Is redirected table equivalent to a view with SECURITY INVOKER?

yes

Reply:
>  This is why i was wondering whether it would need to be added.

I think it should be added. However it has to be added in semantic analyzer code. And it looks like access control check should be added before you make it possible to handle redirected tables here.